aips/aip-103.md

```diff
@@ -0,0 +1,206 @@
+   - Assert $\mathsf{jwt}[\texttt{"nonce"}] \stackrel{?}{=} H’(\mathsf{epk},\mathsf{exp\\_date};\rho)$
+8. Check the EPK expiration date is not too far off into the future:
+   - Assert $\mathsf{exp\\_date} < \mathsf{jwt}[\texttt{"iat"}] + \mathsf{exp\\_horizon}$
+9. Parse $\mathsf{extra\\_field}$ as $\mathsf{extra\\_field\\_key}$ and $\mathsf{extra\\_field\\_val}$ and assert $\mathsf{extra\\_field\\_val}\stackrel{?}{=}\mathsf{jwt}[\mathsf{extra\\_field\\_key}]$
+10. Verify the OIDC signature $\sigma_\mathsf{oidc}$ under $\mathsf{jwk}$ over the JWT $\mathsf{header}$ and payload $\mathsf{jwt}$.
+
+## Alternative solutions
+
+The alternative is to add an additional keyless public key type where the formula to compute the `IdCommitment` does not contain the `aud` at all.
+
+This is the advantage of explicit type safety as a completely new validation path would be implemented.  There would be no risk of such proofs being accepted for accounts that require `aud` to be present due to explicit differences in how the proof would be gated on the type of public key.
+
+However the drawbacks include:
+- We need to add a new keyless public key type, which may not be needed if we can leverage the existing design.  And avoiding proliferation of keyless public key types is desirable.
+- Requiring implementation of a new authentication path in the authenticator, which may be error prone and takes additional engineering effort.
+- Requires more complex changes to the prover as it would need to support a different public inputs hash calculation in order to differentiate between accounts with and without `aud`.  Or it would need to use a different circuit version entirely.
+
+Thus if we can leverage the existing design, it would be preferable to do so.
+
+## Specification and Implementation Details
+
+This AIP's implementation has three parts -
+
+1. We add an additional private input, `skip_aud_check`, into the prover.  This value will indicate whether the `aud` check is enabled.  
+- If it is disabled, the circuit will do the status quo set of verifications.  
+- If it is enabled, the circuit will use an empty string for the `aud` private input (as provided by the prover), which will be used as the `aud` value committed in the IdCommitment.  The circuit will skip matching the value of the JWT's aud claim with the `aud` private input (which is the empty string).  
+
+   Since the `aud` value is committed to in the `IdCommitment`, a proof generated with `skip_aud_check` will be rejected by the validator if the account is not `aud`-less, as the `IdCommitment` will be different, resulting in proof verification failing due to the public inputs hash being incorrect.
+
+   Similarly, a proof generated without `skip_aud_check` will be rejected by the validator if the account *is* `aud`-less, as in this case as well the `IdCommitment` will be different, resulting in proof verification failing due to the public inputs hash being incorrect.
```

Comment:
Same issue here. I would rephrase as:

> Similarly, a proof generated with**out** `skip_aud_check` will be rejected by the validator if the account *is* `aud`-less, since the ZK relation will not be able to match the empty `aud` in the `IdCommitment` with the non-empty `aud` in the JWT.